### PR TITLE
Cluster provisioning part 1

### PR DESCRIPTION
**Pull request by nsricardor**
This pull request implements the provisioning of compute clusters, integrating the latest version of the region service with a shift from physical networks to a more flexible generic network model. While the core provisioning functionality is in place, there are a few features still pending implementation, such as the image selector logic, OpenAPI cluster status support, and the logic to provide custom configuration options post-provisioning.

### CODE REVIEW

**Review by spjmurray — COMMENTED** [review]
Looks relatively small, you're next on my list!

**Review by spjmurray — COMMENTED** [review]
I think other than my pedantry, it's good to test on Monday.  Looking forward to it!

File: pkg/provisioners/managers/cluster/securitygroup.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+	for _, rule := range provisioned {
+		provisionedSet[rule.Metadata.Name] = struct{}{}
+	}
+
+	for _, rule := range desired {
+		desiredSet[rule.ID] = struct{}{}
+	}
+
+	// find rules to delete
+	for id := range provisionedSet {
```
[the remark this answers is not in the document]
Comment (nsricardor) [reply pkg/provisioners/managers/cluster/securitygroup.go:210]: I'll get it updated before your Monday starts ;)

[the remark this answers is not in the document]
Comment (spjmurray) [reply pkg/provisioners/managers/cluster/securitygroup.go:210]: You could even go as far as something like...

```go
type RuleList []Rule

func (l RuleList) IDs() iter.Seq[string] {
  return func(yield func(s string) bool) {
    for _, r := range l {
      return r.ID
    }
  }
}
```

then...

```go
actual := set.New(slices.Collect(rules.IDs())...)
```

some of the new stuff in Go is bloody lovely 😻

**Comment by spjmurray** [comment]
Wing it, I can "create" a cluster in the UI and it says provisioned, I'll add in issues as I encounter them!

**Pull request merged**